File: Tuts/RetainCount/RetainCount/ViewController.m
//
//  ViewController.m
//  RetainCount
//
//  Created by Manoj pratap singh rana on 21/05/17.
//  Copyright © 2017 Impetus. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    NSMutableArray * array = [NSMutableArray new];
    NSLog(@"Retain count is %ld", CFGetRetainCount((__bridge CFTypeRef)array));
    NSString * add1 = [NSString new];
    [array addObject:add1];
    [array addObject:add1];

    NSLog(@"Retain count is %ld", CFGetRetainCount((__bridge CFTypeRef)array));
    NSLog(@"Retain count stringis %ld", CFGetRetainCount((__bridge CFTypeRef)add1));


    
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: SwiftExamples/KeyChainDemo/KeyChainDemo/ViewController.swift
//
//  ViewController.swift
//  KeychainDemo
//
//  Created by Manoj pratap singh rana on 04/04/17.
//  Copyright © 2017 Self Learning. All rights reserved.
//

import UIKit
import Security
class ViewController: UIViewController {

    let myKeychainWrapper = KeychainWrapper(
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        saveKeyInKeyStore()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func saveKeyInKeyStore(){
        //Write values to keychain
        myKeychainWrapper.mySetObject("mmoipihjk", forKey: kSecValueData)
        myKeychainWrapper.writeToKeychain()
        
        // Read values from keychain
       let value =  myKeychainWrapper.myObject(forKey: kSecValueData)as? String
        print(value)
    }
    
    
   /*func listCertificates()
    {
        var status : OSStatus ;
    
        let kSecClassValue = kSecClass.takeRetainedValue() as NSString
        let kSecAttrAccountValue = kSecAttrAccount.takeRetainedValue() as NSString
        let kSecValueDataValue = kSecValueData.takeRetainedValue() as NSString
        let kSecClassGenericPasswordValue = kSecClassGenericPassword.takeRetainedValue() as NSString
        let kSecAttrServiceValue = kSecAttrService.takeRetainedValue() as NSString
        let kSecMatchLimitValue = kSecMatchLimit.takeRetainedValue() as NSString
        let kSecReturnDataValue = kSecReturnData.takeRetainedValue() as NSString
        let kSecMatchLimitOneValue = kSecMatchLimitOne.takeRetainedValue() as NSString
    }*/
}
